server/src/main/java/org/opensearch/http/HttpChunk.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.http;
+
+import org.opensearch.common.annotation.PublicApi;
+import org.opensearch.core.common.bytes.BytesReference;
+
+/**
+ * Represents a chunk of the HTTP response stream
+ */
+@PublicApi(since = "2.11.0")
+public interface HttpChunk {
```

Comment:
This code makes sense, but it's also not used anywhere here so it is hard to comment on it. Do you need to include it now or could this be deferred to subsequent PRs that make use of it?

Reply:
Certainly could be delayed, thanks @andrross , removed it for now